### ISSUE TITLE
Error in left_join(refGeneTxMap, dplyr::rename(refGeneTxMap, newGeneId.merge = newGeneId), : This join would result in more rows than dplyr can handle.

### ISSUE THREAD

**Issue by dudududu12138** [GoekeLab/bambu#429]
Hi，
I used bambu(3.0.8) on my HPC cluster.  I ran with a pacbio hifi sample.  But it reported an error at the quantation step.
Below is my code:
```
#Alignment
minimap2 -ax splice:hq -uf $ref $fastq >result.sam
samtools view -@20 -bS result.sam >resultbam && rm result.sam
samtools sort -@20 -m 4G result.bam -o result.sorted.bam && rm result.bam
#bambu
bambuAnnotations <- prepareAnnotations(ref_anno)
se<-bambu(reads=bam, annotations=bambuAnnotations, genome=ref, ncore=16)
writeBambuOutput(se, path=outpath)
```
Below is the error:
```
--- Start generating read class files ---
Detected 2 warnings across the samples during read class construction. Access warnings with metadata(bambuOutput)$warnings
--- Start extending annotations ---
Using a novel discovery rate (NDR) of: 0.14
--- Start isoform quantification ---
Error: BiocParallel errors
  1 remote errors, element index: 1
  0 unevaluated and other errors
  first remote error:
Error in left_join(refGeneTxMap, dplyr::rename(refGeneTxMap, newGeneId.merge = newGeneId), : This join would result in more rows than dplyr can handle.
```
And the full error report is here [33624781.err.txt](https://github.com/GoekeLab/bambu/files/15466393/33624781.err.txt)

My R version is 4.2.3 and I installed bambu with miniconda3. I firstly create a new environment. The packages in my environment are listed below:
```
# Name                    Version                   Build  Channel
_libgcc_mutex             0.1                 conda_forge    conda-forge
_openmp_mutex             4.5                       2_gnu    conda-forge
_r-mutex                  1.0.1               anacondar_1    conda-forge
_r-xgboost-mutex          2.0                       cpu_0    conda-forge
argcomplete               3.3.0              pyhd8ed1ab_0    conda-forge
binutils_impl_linux-64    2.40                 ha1999f0_1    conda-forge
bioconductor-annotationdbi 1.60.0            r42hdfd78af_0    bioconda
bioconductor-bambu        3.0.8             r42hf17093f_1    bioconda
bioconductor-biobase      2.58.0            r42ha9d7317_1    bioconda
bioconductor-biocfilecache 2.6.0             r42hdfd78af_0    bioconda
bioconductor-biocgenerics 0.44.0            r42hdfd78af_0    bioconda
bioconductor-biocio       1.8.0             r42hdfd78af_0    bioconda
bioconductor-biocparallel 1.32.5            r42hf17093f_1    bioconda
bioconductor-biomart      2.54.0            r42hdfd78af_0    bioconda
bioconductor-biostrings   2.66.0            r42ha9d7317_1    bioconda
bioconductor-bsgenome     1.66.3            r42hdfd78af_0    bioconda
bioconductor-data-packages 20231203             hdfd78af_0    bioconda
bioconductor-delayedarray 0.24.0            r42ha9d7317_1    bioconda
bioconductor-genomeinfodb 1.34.9            r42hdfd78af_0    bioconda
bioconductor-genomeinfodbdata 1.2.9             r42hdfd78af_0    bioconda
bioconductor-genomicalignments 1.34.0            r42ha9d7317_1    bioconda
bioconductor-genomicfeatures 1.50.2            r42hdfd78af_0    bioconda
bioconductor-genomicranges 1.50.0            r42ha9d7317_1    bioconda
bioconductor-iranges      2.32.0            r42ha9d7317_1    bioconda
bioconductor-keggrest     1.38.0            r42hdfd78af_0    bioconda
bioconductor-matrixgenerics 1.10.0            r42hdfd78af_0    bioconda
bioconductor-rhtslib      2.0.0             r42ha9d7317_1    bioconda
bioconductor-rsamtools    2.14.0            r42hf17093f_1    bioconda
bioconductor-rtracklayer  1.58.0            r42h58c1800_2    bioconda
bioconductor-s4vectors    0.36.0            r42ha9d7317_1    bioconda
bioconductor-summarizedexperiment 1.28.0            r42hdfd78af_0    bioconda
bioconductor-xvector      0.38.0            r42ha9d7317_1    bioconda
bioconductor-zlibbioc     1.44.0            r42ha9d7317_2    bioconda
bwidget                   1.9.14               ha770c72_1    conda-forge
bzip2                     1.0.8                hd590300_5    conda-forge
c-ares                    1.28.1               hd590300_0    conda-forge
ca-certificates           2024.2.2             hbcca054_0    conda-forge
cairo                     1.18.0               h3faef2a_0    conda-forge
curl                      8.8.0                he654da7_0    conda-forge
expat                     2.6.2                h59595ed_0    conda-forge
font-ttf-dejavu-sans-mono 2.37                 hab24e00_0    conda-forge
font-ttf-inconsolata      3.000                h77eed37_0    conda-forge
font-ttf-source-code-pro  2.038                h77eed37_0    conda-forge
font-ttf-ubuntu           0.83                 h77eed37_2    conda-forge
fontconfig                2.14.2               h14ed4e7_0    conda-forge
fonts-conda-ecosystem     1                             0    conda-forge
fonts-conda-forge         1                             0    conda-forge
freetype                  2.12.1               h267a509_2    conda-forge
fribidi                   1.0.10               h36c2ea0_0    conda-forge
gcc_impl_linux-64         13.2.0               h9eb54c0_7    conda-forge
gfortran_impl_linux-64    13.2.0               h738fc78_7    conda-forge
graphite2                 1.3.13            h59595ed_1003    conda-forge
gxx_impl_linux-64         13.2.0               h2a599c4_7    conda-forge
harfbuzz                  8.5.0                hfac3d4d_0    conda-forge
icu                       73.2                 h59595ed_0    conda-forge
jq                        1.7.1                hd590300_0    conda-forge
kernel-headers_linux-64   2.6.32              he073ed8_17    conda-forge
keyutils                  1.6.1                h166bdaf_0    conda-forge
krb5                      1.21.2               h659d440_0    conda-forge
ld_impl_linux-64          2.40                 hf3520f5_1    conda-forge
lerc                      4.0.0                h27087fc_0    conda-forge
libblas                   3.9.0           22_linux64_openblas    conda-forge
libcurl                   8.8.0                hca28451_0    conda-forge
libdeflate                1.20                 hd590300_0    conda-forge
libedit                   3.1.20191231         he28a2e2_2    conda-forge
libev                     4.33                 hd590300_2    conda-forge
libexpat                  2.6.2                h59595ed_0    conda-forge
libffi                    3.4.2                h7f98852_5    conda-forge
libgcc-devel_linux-64     13.2.0             hceb6213_107    conda-forge
libgcc-ng                 13.2.0               h77fa898_7    conda-forge
libgfortran-ng            13.2.0               h69a702a_7    conda-forge
libgfortran5              13.2.0               hca663fb_7    conda-forge
libglib                   2.80.2               hf974151_0    conda-forge
libgomp                   13.2.0               h77fa898_7    conda-forge
libiconv                  1.17                 hd590300_2    conda-forge
libjpeg-turbo             3.0.0                hd590300_1    conda-forge
liblapack                 3.9.0           22_linux64_openblas    conda-forge
libnghttp2                1.58.0               h47da74e_1    conda-forge
libnsl                    2.0.1                hd590300_0    conda-forge
libopenblas               0.3.27          pthreads_h413a1c8_0    conda-forge
libpng                    1.6.43               h2797004_0    conda-forge
libsanitizer              13.2.0               h6ddb7a1_7    conda-forge
libsqlite                 3.45.3               h2797004_0    conda-forge
libssh2                   1.11.0               h0841786_0    conda-forge
libstdcxx-devel_linux-64  13.2.0             hceb6213_107    conda-forge
libstdcxx-ng              13.2.0               hc0a3c3a_7    conda-forge
libtiff                   4.6.0                h1dd3fc0_3    conda-forge
libuuid                   2.38.1               h0b41bf4_0    conda-forge
libwebp-base              1.4.0                hd590300_0    conda-forge
libxcb                    1.15                 h0b41bf4_0    conda-forge
libxcrypt                 4.4.36               hd590300_1    conda-forge
libxgboost                2.0.3            cpu_h7afb2cf_4    conda-forge
libxml2                   2.12.7               hc051c1a_0    conda-forge
libzlib                   1.2.13               hd590300_5    conda-forge
make                      4.3                  hd18ef5c_1    conda-forge
ncurses                   6.5                  h59595ed_0    conda-forge
oniguruma                 6.9.9                hd590300_0    conda-forge
openssl                   3.3.0                h4ab18f5_3    conda-forge
pango                     1.52.2               ha41ecd1_0    conda-forge
pcre2                     10.43                hcad00b1_0    conda-forge
pip                       24.0               pyhd8ed1ab_0    conda-forge
pixman                    0.43.2               h59595ed_0    conda-forge
pthread-stubs             0.4               h36c2ea0_1001    conda-forge
python                    3.12.3          hab00c5b_0_cpython    conda-forge
python_abi                3.12                    4_cp312    conda-forge
pyyaml                    6.0.1           py312h98912ed_1    conda-forge
r-askpass                 1.2.0             r42h57805ef_0    conda-forge
r-assertthat              0.2.1             r42hc72bb7e_4    conda-forge
r-base                    4.2.3               hcf4f30c_13    conda-forge
r-bh                      1.84.0_0          r42hc72bb7e_0    conda-forge
r-bit                     4.0.5             r42h57805ef_1    conda-forge
r-bit64                   4.0.5             r42h57805ef_2    conda-forge
r-bitops                  1.0_7             r42h57805ef_2    conda-forge
r-blob                    1.2.4             r42hc72bb7e_1    conda-forge
r-cachem                  1.1.0             r42hb1dbf0f_0    conda-forge
r-cli                     3.6.2             r42ha503ecb_0    conda-forge
r-codetools               0.2_20            r42hc72bb7e_0    conda-forge
r-cpp11                   0.4.7             r42hc72bb7e_0    conda-forge
r-crayon                  1.5.2             r42hc72bb7e_2    conda-forge
r-curl                    5.1.0             r42hf9611b0_0    conda-forge
r-data.table              1.15.2            r42h029312a_0    conda-forge
r-dbi                     1.2.2             r42hc72bb7e_0    conda-forge
r-dbplyr                  2.5.0             r42hc72bb7e_0    conda-forge
r-digest                  0.6.35            r42ha503ecb_0    conda-forge
r-dplyr                   1.1.4             r42ha503ecb_0    conda-forge
r-ellipsis                0.3.2             r42h57805ef_2    conda-forge
r-fansi                   1.0.6             r42h57805ef_0    conda-forge
r-fastmap                 1.2.0             r42ha18555a_0    conda-forge
r-filelock                1.0.3             r42h57805ef_0    conda-forge
r-formatr                 1.14              r42hc72bb7e_1    conda-forge
r-futile.logger           1.4.3           r42hc72bb7e_1005    conda-forge
r-futile.options          1.0.1           r42hc72bb7e_1004    conda-forge
r-generics                0.1.3             r42hc72bb7e_2    conda-forge
r-glue                    1.7.0             r42h57805ef_0    conda-forge
r-hms                     1.1.3             r42hc72bb7e_1    conda-forge
r-httr                    1.4.7             r42hc72bb7e_0    conda-forge
r-jsonlite                1.8.8             r42h57805ef_0    conda-forge
r-lambda.r                1.2.4             r42hc72bb7e_3    conda-forge
r-lattice                 0.22_6            r42h57805ef_0    conda-forge
r-lifecycle               1.0.4             r42hc72bb7e_0    conda-forge
r-magrittr                2.0.3             r42h57805ef_2    conda-forge
r-matrix                  1.6_5             r42h316c678_0    conda-forge
r-matrixstats             1.3.0             r42h57805ef_0    conda-forge
r-memoise                 2.0.1             r42hc72bb7e_2    conda-forge
r-mime                    0.12              r42h57805ef_2    conda-forge
r-openssl                 2.2.0             r42hd7d0243_0    conda-forge
r-pillar                  1.9.0             r42hc72bb7e_1    conda-forge
r-pkgconfig               2.0.3             r42hc72bb7e_3    conda-forge
r-plogr                   0.2.0           r42hc72bb7e_1005    conda-forge
r-png                     0.1_8             r42h81d01c5_1    conda-forge
r-prettyunits             1.2.0             r42hc72bb7e_0    conda-forge
r-progress                1.2.3             r42hc72bb7e_0    conda-forge
r-purrr                   1.0.2             r42h57805ef_0    conda-forge
r-r6                      2.5.1             r42hc72bb7e_2    conda-forge
r-rappdirs                0.3.3             r42h57805ef_2    conda-forge
r-rcpp                    1.0.12            r42h7df8631_0    conda-forge
r-rcpparmadillo           0.12.8.3.0        r42h58a4165_0    conda-forge
r-rcurl                   1.98_1.14         r42hf9611b0_0    conda-forge
r-restfulr                0.0.15            r42h56115f1_2    bioconda
r-rjson                   0.2.21            r42ha503ecb_3    conda-forge
r-rlang                   1.1.3             r42ha503ecb_0    conda-forge
r-rsqlite                 2.3.4             r42ha503ecb_0    conda-forge
r-snow                    0.4_4             r42hc72bb7e_2    conda-forge
r-stringi                 1.8.4             r42hbd1cc82_0    conda-forge
r-stringr                 1.5.1             r42h785f33e_0    conda-forge
r-sys                     3.4.2             r42h57805ef_1    conda-forge
r-tibble                  3.2.1             r42h57805ef_2    conda-forge
r-tidyr                   1.3.1             r42ha503ecb_0    conda-forge
r-tidyselect              1.2.1             r42hc72bb7e_0    conda-forge
r-utf8                    1.2.4             r42h57805ef_0    conda-forge
r-vctrs                   0.6.5             r42ha503ecb_0    conda-forge
r-withr                   3.0.0             r42hc72bb7e_0    conda-forge
r-xgboost                 2.0.3           cpu_r42h5cb4a28_4    conda-forge
r-xml                     3.99_0.16.1       r42hc6530ce_0    conda-forge
r-xml2                    1.3.6             r42hbfba7a4_1    conda-forge
r-yaml                    2.3.8             r42h57805ef_0    conda-forge
readline                  8.2                  h8228510_1    conda-forge
sed                       4.8                  he412f7d_0    conda-forge
setuptools                70.0.0             pyhd8ed1ab_0    conda-forge
sysroot_linux-64          2.12                he073ed8_17    conda-forge
tk                        8.6.13          noxft_h4845f30_101    conda-forge
tktable                   2.10                 h0c5db8f_5    conda-forge
toml                      0.10.2             pyhd8ed1ab_0    conda-forge
tomlkit                   0.12.5             pyha770c72_0    conda-forge
tzdata                    2024a                h0c530f3_0    conda-forge
wheel                     0.43.0             pyhd8ed1ab_1    conda-forge
xmltodict                 0.13.0             pyhd8ed1ab_0    conda-forge
xorg-kbproto              1.0.7             h7f98852_1002    conda-forge
xorg-libice               1.1.1                hd590300_0    conda-forge
xorg-libsm                1.2.4                h7391055_0    conda-forge
xorg-libx11               1.8.9                h8ee46fc_0    conda-forge
xorg-libxau               1.0.11               hd590300_0    conda-forge
xorg-libxdmcp             1.1.3                h7f98852_0    conda-forge
xorg-libxext              1.3.4                h0b41bf4_2    conda-forge
xorg-libxrender           0.9.11               hd590300_0    conda-forge
xorg-libxt                1.3.0                hd590300_1    conda-forge
xorg-renderproto          0.11.1            h7f98852_1002    conda-forge
xorg-xextproto            7.3.0             h0b41bf4_1003    conda-forge
xorg-xproto               7.0.31            h7f98852_1007    conda-forge
xz                        5.2.6                h166bdaf_0    conda-forge
yaml                      0.2.5                h7f98852_2    conda-forge
yq                        3.4.3              pyhd8ed1ab_0    conda-forge
zlib                      1.2.13               hd590300_5    conda-forge
zstd                      1.5.6                ha6fb4c9_0    conda-forge
```

**Comment by andredsim** [comment]
Hi sorry for the delayed response,

Thank you for running Bambu, let's see if we can solve the issue. I havn't see this error before so I am not sure what the solution is yet. I cannot see anything wrong with how you are running Bambu so my first avenues to troubleshoot would be the Bambu version and the input.
1. Version 3.0.8 is an older version of Bambu. Could you update to the latest by downloading it straight from github? Or use bioconductor with Rv 4.4? And see if the problem persists.
2. The error seems to be caused by a very high number of novel genes, this is quite atypical in most samples. Is there anything non-standard about your input data? Which organism is it? How well annotated is it? Was your library prep standard, or use selection? Do you expect lots of scattered or overlapping reads? How large is your input bam file?

Let me know and I can continue to troubleshoot from there?
Kind Regards,
Andre Sim

**Comment by dudududu12138** [comment]
> Hi sorry for the delayed response,
> 
> Thank you for running Bambu, let's see if we can solve the issue. I havn't see this error before so I am not sure what the solution is yet. I cannot see anything wrong with how you are running Bambu so my first avenues to troubleshoot would be the Bambu version and the input.
> 
> 1. Version 3.0.8 is an older version of Bambu. Could you update to the latest by downloading it straight from github? Or use bioconductor with Rv 4.4? And see if the problem persists.
> 2. The error seems to be caused by a very high number of novel genes, this is quite atypical in most samples. Is there anything non-standard about your input data? Which organism is it? How well annotated is it? Was your library prep standard, or use selection? Do you expect lots of scattered or overlapping reads? How large is your input bam file?
> 
> Let me know and I can continue to troubleshoot from there? Kind Regards, Andre Sim

Hi,thanks for your reply. I have solved the problem by updating the version to 4.4.

**Comment by andredsim** [comment]
Glad to hear it was a simple fix! Best of luck with the rest of your analysis
